lib/nexpose/site.rb

```diff
@@ -203,9 +203,50 @@ def add_ip(ip)
     #
     # @param [String] ip IP address of an asset.
     def remove_ip(ip)
-      @assets = assets.reject { |asset| asset == IPRange.new(ip) }
+      ip = IPRange.new(ip)
+      @assets.each do |asset_range|
+        next if asset_range.is_a?(Nexpose::HostName)
+        if asset_range.include?(ip)
+          asset = split_ip_range(asset_range, ip)
+          @assets.delete(asset_range)
+          @assets.push(asset)
+          @assets.flatten!
```

Comment:
Instead of calling `Array#flatten!`, you can use `Array#+` which keeps things one-dimensional:

```ruby
assets += Array(asset)
```